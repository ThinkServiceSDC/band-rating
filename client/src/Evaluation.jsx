import React, { Component } from 'react';
import './evaluation.css';
import sendEvaluation from './backendCalls';

export default class Evaluation extends Component {
    state = { bandName: '', vote: null, comment: '' };

    constructor(props) {
      super(props);
      this.updateVote = this.updateVote.bind(this);
      this.evaluateBand = this.evaluateBand.bind(this);
    }

    updateVote(event) {
      this.setState({ vote: event.target.value });
    }

    async evaluateBand(event) {
      const { bandName, vote, comment } = this.state;

      event.preventDefault();
      sendEvaluation(bandName, vote, comment);
    }

    render() {
      const { bandName, comment } = this.state;

      return (
        <div className="evaluation-form">
          <form id="submitForm" onSubmit={this.evaluateBand}>
            <div className="evaluation-container">
              <input
                id="bandName"
                className="insert-field"
                type="text"
                name="bandName"
                value={bandName}
                placeholder="Band"
                onChange={event => this.setState({ bandName: event.target.value })}
              />
              <div className="voting-area">
                <div className="radio-button">
                  <label htmlFor="Good" className="radio-button-label">
                    <input id="Good" type="radio" value="Good" name="rating" onChange={this.updateVote} />
                    Good
                  </label>
                </div>
                <div className="radio-button">
                  <label htmlFor="Medium" className="radio-button-label">
                    <input
                      type="radio"
                      id="Medium"
                      value="Medium"
                      name="rating"
                      onChange={this.updateVote}
                    />
                                    Medium
                  </label>
                </div>
                <div className="radio-button">
                  <label htmlFor="Bad" className="radio-button-label">
                    <input
                      type="radio"
                      id="Bad"
                      value="Bad"
                      name="rating"
                      onChange={this.updateVote}
                    />
                    Bad
                  </label>
                </div>
                <div className="radio-button">
                  <label htmlFor="Special" className="radio-button-label">
                    <input
                      type="radio"
                      id="Special"
                      value="Special"
                      name="rating"
                      onChange={this.updateVote}
                    />
                    Special
                  </label>
                </div>
                <div className="radio-button">
                  <label htmlFor="MetalBattle" className="radio-button-label">
                    <input
                      type="radio"
                      id="MetalBattle"
                      value="MetalBattle"
                      name="rating"
                      onChange={this.updateVote}
                    />
                    Metal Battle
                  </label>
                </div>
              </div>
              <input
                id="comment"
                className="insert-field"
                type="text"
                name="comment"
                value={comment}
                placeholder="Comment"
                onChange={event => this.setState({ comment: event.target.value })}
              />
              <input id="Evaluate" className="submit" type="submit" value="Evaluate" />
            </div>
          </form>
        </div>
      );
    }
}
