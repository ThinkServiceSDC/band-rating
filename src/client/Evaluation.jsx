import React, {Component} from 'react';
import './evaluation.css';

export default class Evaluation extends Component {

    constructor(props) {
        super(props);
        this.state = {selectedVote: null};
    }

    render() {
        return (
            <div className='evaluation-form'>
                <form onSubmit={this.evaluateBand}>
                    <div className='evaluation-container'>
                        <input className='band-insert-field' type='text' name='bandName'/>
                        <div className='voting-area'>
                            <div className='radio-button'>
                                <input type='radio' id='Good' value='Good' name='rating' onChange={this.voteChanged}/>
                                <label className='radio-button-label'>Good</label>
                            </div>
                            <div className='radio-button'>
                                <input type='radio' id='Medium' value='Medium' name='rating'
                                       onChange={this.voteChanged}/>
                                <label className='radio-button-label'>Medium</label>
                            </div>
                            <div className='radio-button'>
                                <input type='radio' id='Bad' value='Bad' name='rating' onChange={this.voteChanged}/>
                                <label className='radio-button-label'>Bad</label>
                            </div>
                            <div className='radio-button'>
                                <input type='radio' id='Special' value='Special' name='rating'
                                       onChange={this.voteChanged}/>
                                <label className='radio-button-label'>Special</label>
                            </div>
                            <div className='radio-button'>
                                <input type='radio' id='MetalBattle' value='MetalBattle' name='rating'
                                       onChange={this.voteChanged}/>
                                <label className='radio-button-label'>Metal Battle</label>
                            </div>
                        </div>
                        <input className='submit' type='submit' value='Evaluate'/>
                    </div>
                </form>
            </div>
        )
    }

    voteChanged = (event) => {
        this.setState({selectedVote: event.target.value});
    };

    evaluateBand = async () => {
        event.preventDefault();
        console.log(this.state.selectedVote);
    };
}