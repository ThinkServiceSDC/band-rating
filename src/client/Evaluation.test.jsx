import React from 'react';
import toJSON from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Evaluation from "./Evaluation";

describe('Evaluation', () => {
    it('should render correctly', () => {
        expect(toJSON(shallow(<Evaluation/>))).toMatchSnapshot();
    });

    it('should change state when selecting different option', () => {
        const wrapper = shallow(<Evaluation/>);
        wrapper.find('#Good').simulate('change', {target: {value: 'Good'}});
        expect(wrapper.state()).toEqual({selectedVote: 'Good'});
    });
});