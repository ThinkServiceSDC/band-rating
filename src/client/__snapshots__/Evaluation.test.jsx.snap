// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evaluation should render correctly 1`] = `
<div
  className="evaluation-form"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="evaluation-container"
    >
      <input
        className="band-insert-field"
        name="bandName"
        type="text"
      />
      <div
        className="voting-area"
      >
        <div
          className="radio-button"
        >
          <input
            id="Good"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="Good"
          />
          <label
            className="radio-button-label"
          >
            Good
          </label>
        </div>
        <div
          className="radio-button"
        >
          <input
            id="Medium"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="Medium"
          />
          <label
            className="radio-button-label"
          >
            Medium
          </label>
        </div>
        <div
          className="radio-button"
        >
          <input
            id="Bad"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="Bad"
          />
          <label
            className="radio-button-label"
          >
            Bad
          </label>
        </div>
        <div
          className="radio-button"
        >
          <input
            id="Special"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="Special"
          />
          <label
            className="radio-button-label"
          >
            Special
          </label>
        </div>
        <div
          className="radio-button"
        >
          <input
            id="MetalBattle"
            name="rating"
            onChange={[Function]}
            type="radio"
            value="MetalBattle"
          />
          <label
            className="radio-button-label"
          >
            Metal Battle
          </label>
        </div>
      </div>
      <input
        className="submit"
        type="submit"
        value="Evaluate"
      />
    </div>
  </form>
</div>
`;
